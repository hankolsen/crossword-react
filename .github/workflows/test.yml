name: melodikryzzet-tests

on: 
  push:
    branches:
      - 'master'
  pull_request:


jobs:
  test:
    name: 'Node ${{ matrix.node }}'
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node: [12, 14]
    env:
      SKIP_PREFLIGHT_CHECK: true
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v1
      with: 
        node-version: ${{ matrix.node }}
    - name: Install dependencies
      uses: bahmutov/npm-install@v1
      with: 
        install-command: yarn
    - name: Run tests
      run: yarn test:ci