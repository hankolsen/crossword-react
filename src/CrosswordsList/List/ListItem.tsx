import React from 'react';
import { Link } from 'react-router-dom';

type Props = {
  crossword: {
    id: string;
    name: string;
  };
};

const ListItem = (props: Props) => {
  const {
    crossword: { id, name },
  } = props;

  return (
    <div className="list__list-item">
      <Link className="list-item__button" to={`/crossword/${id}`} key={id}>
        {name}
      </Link>
    </div>
  );
};

export default ListItem;
