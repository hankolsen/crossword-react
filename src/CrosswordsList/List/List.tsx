import React from 'react';
import ListItem from './ListItem';
import './List.css';

type Props = {
  crosswords: {
    id: string;
    name: string;
  }[];
};

const List = ({ crosswords }: Props) => (
  <div className="list">
    {crosswords.map((crossword) => (
      <ListItem crossword={crossword} key={crossword.id} />
    ))}
  </div>
);

export default List;
