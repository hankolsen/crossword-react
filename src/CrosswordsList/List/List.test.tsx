import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from 'test/test-utils';
import List from './List';

describe('List test', () => {
  it('should render a list item', () => {
    const crosswords = [
      {
        id: 'abc-123',
        name: 'The Crossowrd to rule them all',
      },
      {
        id: 'cde-456',
        name: 'The worst crossowrd',
      },
    ];
    const { container } = render(
      <BrowserRouter>
        <List crosswords={crosswords} />
      </BrowserRouter>,
    );
    expect(container.querySelectorAll('a').length).toBe(2);
    expect(container.querySelectorAll('a').item(1).href).toBe(
      'http://localhost/crossword/cde-456',
    );
    expect(container).toMatchSnapshot();
  });
});
