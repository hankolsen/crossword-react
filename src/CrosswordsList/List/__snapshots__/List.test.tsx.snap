// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List test should render a list item 1`] = `
<div>
  <div>
    <div
      class="list"
    >
      <div
        class="list__list-item"
      >
        <a
          class="list-item__button"
          href="/crossword/abc-123"
        >
          The Crossowrd to rule them all
        </a>
      </div>
      <div
        class="list__list-item"
      >
        <a
          class="list-item__button"
          href="/crossword/cde-456"
        >
          The worst crossowrd
        </a>
      </div>
    </div>
  </div>
</div>
`;
